File: tests/fixtures/ts/rebinding/src/destructured.ts
import Stripe from 'stripe';

const stripe = new Stripe(process.env.STRIPE_KEY!);

export async function indexed(rows: any[]) {
  const charge = await stripe.charges.create({ amount: 100, currency: 'usd' });

  let described;
  {
    const { charge } = rows[0];
    described = charge.leaked;
  }

  return { described, state: charge.status };
}

File: tests/fixtures/py/rebinding/src/destructured.py
from stripe import StripeClient

client = StripeClient("sk_test")


def indexed(rows):
    charge = client.charges.create(amount=100, currency="usd")

    def split():
        head, (charge, tail) = rows
        return head, charge.leaked, tail

    return split(), charge.status
